### PR TITLE
doc: Add FAQ page

### PR DESCRIPTION
**Pull request by abhinav**
Adds an FAQ page to the Fx documentation,
aiming to address frequently asked questions.

This includes two entries that came up at work recently.

Includes a test for the `fx.Supply` entry
because I couldn't find the test for it right away.
(I found it in annotated_test.go later but you won't hear me complain
about additional tests.)

### CODE REVIEW

File: docs/faq.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
+# Frequently Asked Questions
+
+This page contains answers to common questions and issues with using Fx.
+
+## Does the order of `fx.Option`s matter?
+
+No, the order in which you provide various Fx options
+to `fx.Options`, `fx.New`, `fx.Module`, and others does not matter.
```
Comment (JacobOaks) [inline docs/faq.md:8]: I don't think this is true for determining the order in which `fx.Invoke`d functions get run, which was a recent point of confusion (ref: https://github.com/uber-go/fx/issues/1152). Consider exempting Invoke from this statement.

**Review by JacobOaks — APPROVED** [review]
Awesome, thanks! One small comment.

File: docs/faq.md (hunk 1 above)
Comment (abhinav) [reply docs/faq.md:8]: Fair. There's a weird situation here.
On paper, there's no guarantee about the order in which invokes run.
But you're right, their order matters.

Comment (abhinav) [reply docs/faq.md:8]: @JacobOaks I realized that this was actually too simplified and lost some information.
Ordering of options relative to fx.New is irrelevant, but ordering of operations relative to each other can matter.
I've rewritten this section to better clarify this.

LMK what you think

Comment (abhinav) [reply docs/faq.md:8]: Actually, looks like we do document the fx.Invoke invocation order.
I'll update this entry.

**Review by JacobOaks — APPROVED** [review]
👍 thanks!

**Pull request merged**